src/hooks/resources/device.ts

```diff
@@ -572,3 +585,58 @@ async function checkSupervisorReleaseUpgrades(
 		}
 	}
 }
+
+async function setSupervisorReleaseResource(
+	api: sbvrUtils.PinejsClient,
+	deviceIds: number[],
+	supervisorVersion: string,
+) {
+	// This function _should_ get called with one device (since each supervisor reports itself)
+	const device = (await api.get({
+		resource: 'device',
+		id: deviceIds[0],
+		options: {
+			$expand: ['is_of__device_type'],
+		},
+	})) as AnyObject;
+
+	if (device.supervisor_version != null) {
+		// if the device already has a supervisor_version, just bail.
+		return null;
+	}
+
+	const supervisorRelease = (await api.get({
+		resource: 'supervisor_release',
+		options: {
+			$select: ['id'],
+			$filter: {
+				supervisor_version: `v${supervisorVersion}`,
+				is_for__device_type: {
+					$any: {
+						$alias: 'dt',
+						$expr: {
+							dt: {
+								id: device.is_of__device_type[0].id,
+							},
+						},
+					},
+				},
+			},
+		},
+	})) as AnyObject[];
+
+	if (supervisorRelease == null) {
+		throw new BadRequestError(
+			// TODO: maybe this shouldn't throw (for the oB case with empty supervisor_releases)
+			`Could not find a supervisor release for this device type & version (${device.is_of__device_type}, ${supervisorVersion})`,
+		);
+	}
+
+	return await api.patch({
```

Comment:
I think this can be handled by adding
```ts
request.values.should_be_managed_by__supervisor_release = supervisorRelease[0].id
```
and adding it to the current patch that's running

Reply:
How can I add this bit to the current patch? Is that something better suited for a `POSTPARSE` hook to reuse the SQL compile step?